lost/src/main/java/com/mapzen/android/lost/internal/GeofencingApiImpl.java

```diff
@@ -16,32 +17,57 @@
 public class GeofencingApiImpl implements GeofencingApi {
 
   private final LocationManager locationManager;
+  private HashMap<String, PendingIntent> pendingIntentMap;
 
   GeofencingApiImpl(Context context) {
     locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+    pendingIntentMap = new HashMap<>();
   }
 
   @Override
   public void addGeofences(GeofencingRequest geofencingRequest, PendingIntent pendingIntent)
       throws SecurityException {
-    ParcelableGeofence geofence = (ParcelableGeofence) geofencingRequest.getGeofences().get(0);
-    locationManager.addProximityAlert(
-        geofence.getLatitude(),
-        geofence.getLongitude(),
-        geofence.getRadius(),
-        geofence.getDuration(),
-        pendingIntent);
+    List<Geofence> geofences = geofencingRequest.getGeofences();
+    addGeofences(geofences, pendingIntent);
   }
 
-  @Override public void addGeofences(List<Geofence> geofences, PendingIntent pendingIntent) {
-    throw new RuntimeException("Sorry, not yet implemented");
+  @Override public void addGeofences(List<Geofence> geofences, PendingIntent pendingIntent)
+      throws SecurityException {
+    for (Geofence geofence : geofences) {
+      addGeofence(geofence, pendingIntent);
+    }
+  }
+
+  private void addGeofence(Geofence geofence, PendingIntent pendingIntent)
```

Comment:
It looks like with build tools 24 adding `throws SecurityException` to the method signature is no longer sufficient to satisfy static analysis in Android Studio as this method's implementation is still highlighted in red.

It seems our only option is to start using `@RequiresPermission(anyOf = {ACCESS_COARSE_LOCATION, ACCESS_FINE_LOCATION})` on internal methods as well to suppress these warnings.